app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -94,10 +116,34 @@ class StateFragmentPresenter @Inject constructor(
 
   private fun subscribeToCurrentState() {
     ephemeralStateLiveData.observe(fragment, Observer<EphemeralState> { result ->
+      if (result.state.interaction.id.equals("NumericInput"))
+        addNumberInputContentCard(result.state.interaction.customizationArgsMap.get("placeholder")!!.normalizedString)
       logger.d("StateFragment", "getCurrentState: ${result.state.name}")
     })
   }
 
+  /** The function for adding [NumberInputInteractionView]. */
+  fun addNumberInputContentCard(placeholder: String) {
+    contentComponent = NumberInputInteractionView(
+      fragment.context!!,
+      placeholder,
+      numberOfrows
+    )
+    val params = LinearLayout.LayoutParams(
+      LinearLayout.LayoutParams.MATCH_PARENT,
+      LinearLayout.LayoutParams.WRAP_CONTENT
+    )
+    params.setMargins(dpToPx(8), dpToPx(8), dpToPx(8), dpToPx(8))
+    llRoot.addView(contentComponent, params)
+    contentComponent.setText(digit)
+    if(isFetched)
+      dummyFetchDataTV.setText(digit)
+    dummyEditTextButton!!.setOnClickListener(View.OnClickListener {
```

Comment:
Indentation incorrect. Also remove `!!` from here too.

Reply:
Indentation is proper , to make the code review easier adding curly brackets.